### ISSUE TITLE
Verification Error steady 

### ISSUE THREAD

**Issue by thanosvlo** [jakeret/tf_unet#134]
Hi, Great work

One question, I'm training on a custom dataset and i have noticed that the verification error stays the same throughout the training even though the loss is decreasing. Do you have any insight why that would be the case ? 
 
net = unet.Unet(layers=5, features_root=32, channels=3, n_class=2)
trainer = unet.Trainer(net,optimizer="adam",batch_size=32,opt_kwargs=dict(beta1=0.91,learning_rate=0.00001))
path = trainer.train(data_provider, output_path, training_iters=574, epochs=800)

this is how I initialise  my net 

2017-11-14 12:41:05,737 Epoch 3, Average loss: 0.4687, learning rate: 0.0000
2017-11-14 12:41:05,815 Verification error= 13.3%, loss= 0.5009

2017-11-14 13:52:03,107 Epoch 5, Average loss: 0.4829, learning rate: 0.0000
2017-11-14 13:52:03,185 Verification error= 13.3%, loss= 0.4730

( I do realize this example is taken from very close epochs but just exhibiting a point , also the lr is not 0 its 0.00001 ) 
Cheers

**Comment by jakeret** [comment]
hard to tell. Is the stored prediction also remaining the same?

**Comment by thanosvlo** [comment]
Nope that actually is getting better with more training, so the net is actually learning. 

Could it be the small sample that is used for verification ? If i remember correctly you have hardcoded to get 4 images for verification 

Could that affect the training results ?

**Comment by huang229** [comment]
@thanosvlo Have you solved this problem, please?

**Comment by thanosvlo** [comment]
@huang229 No, I ended up building my own unet
